mountpoint-s3-client/src/object_client.rs

```diff
@@ -254,6 +254,8 @@ pub enum GetObjectAttributesError {
 pub struct PutObjectParams {
     /// Enable Crc32c trailing checksums.
     pub trailing_checksums: bool,
+    /// Storage class to be used when creating new S3 object
+    pub storage_class: String,
```

Comment:
Does this work if you use `PutObjectParams::default()`, which will give an empty string? I don't know if the empty string is a valid argument to `x-amz-storage-class`. My guess is we probably want this to be `Option<String>` and only include the header when it's `Some`. For Mountpoint we'll always set it to `Some`, but other users of this `mountpoint-s3-client` crate might have other preferences or no preference.